docs/use/transform.md

```diff
@@ -0,0 +1,139 @@
+# Transforming specifications
+
+The `swagger` toolkit allows some transformation to be carried on with a specification.
+
+Currently it is possible to apply the following transforms:
+- expansion: this expands all `$ref`'s in a spec
+- minimal flattening: carries on minimal transformation to a spec to be workable for the swagger codegen
+- full flattening: performs minimal flattening and in addition, replaces all complex constructs in schemas by named definitions
+- mixin: merges one or more specifications into a primary spec
+
+### Expansion
+
+Expanding a spec may prove useful to produce documentation or test cases. The primary intent is not code generation.
+
+> **NOTE**: Circular `$ref` are detected and remain as local `$ref`
+
+Usage:
+
+`swagger expand {spec}`
+or
+
+`swagger flatten --with-expand` {spec}`
+
+or with codegen commands:
+
+`swagger generate [model|server|client|operation|...] --spec={spec} --with-expand`
+
+> **NOTE**: codegen may fail in some situations with spec expansion:
+> - polymorphism: the original intent of the `$ref` to a discriminated type(i.e. _extends type..._) is lost with expansion
+> - duplicate names: expansion of `$ref` intended for type reuse may produce duplicate type identifiers
+> - too many generated files: expansion of large specs may lead to many independent anonymous structures
+
+### Minimal flattening
+
+This transforms makes complex JSON `$ref` amenable to analysis and code generation.
```

Comment:
"This transform makes" or "These transforms make"

Reply:
done